server/sonar-db-dao/src/test/java/org/sonar/db/rule/RuleTesting.java

```diff
@@ -46,60 +50,128 @@ private RuleTesting() {
     // only static helpers
   }
 
+  public static RuleDefinitionDto newRule() {
+    RuleKey key = RuleKey.of("repo_" + randomAlphanumeric(5), "key_" + randomAlphanumeric(5));
+    return newRule(key);
+  }
+
+  public static RuleDefinitionDto newRule(RuleKey key) {
+    return new RuleDefinitionDto()
+      .setRepositoryKey(key.repository())
+      .setRuleKey(key.rule())
+      .setName("name_" + randomAlphanumeric(5))
+      .setDescription("description_" + randomAlphanumeric(5))
+      .setDescriptionFormat(Format.HTML)
+      .setType(RuleType.values()[nextInt(RuleType.values().length)])
+      .setStatus(RuleStatus.READY)
+      .setConfigKey("configKey_" + randomAlphanumeric(5))
+      .setSeverity(Severity.ALL.get(nextInt(Severity.ALL.size())))
+      .setIsTemplate(false)
+      .setSystemTags(newHashSet("tag_" + randomAlphanumeric(5), "tag_" + randomAlphanumeric(5)))
+      .setLanguage("lang_" + randomAlphanumeric(3))
+      .setGapDescription("gapDescription_" + randomAlphanumeric(5))
+      .setDefRemediationBaseEffort(nextInt(10) + "h")
+      .setDefRemediationGapMultiplier(nextInt(10) + "h")
+      .setDefRemediationFunction("LINEAR_OFFSET")
+      .setCreatedAt(System.currentTimeMillis())
+      .setUpdatedAt(System.currentTimeMillis());
+  }
+
+  public static RuleMetadataDto newRuleMetadata() {
+    return randomizeFields(new RuleMetadataDto()
+      .setRuleId(nextInt(1_000_000))
+      .setOrganizationUuid("org_" + randomAlphanumeric(5)));
+  }
+
+  public static RuleMetadataDto newRuleMetadata(RuleDefinitionDto rule, OrganizationDto organization) {
+    return randomizeFields(new RuleMetadataDto()
+      .setRuleId(rule.getId())
+      .setOrganizationUuid(organization.getUuid()));
```

Comment:
I would suggest to always randomize all fields and afterwards overwrite the required fields.